### PR TITLE
First_Commit

### PR DESCRIPTION
**Pull request by chetanphadke3**
This PR is for opening the OrangeHRM Login page only.

### CODE REVIEW

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+
+
+//Selenium Wrapper Class
+public class PredefinedMethods {
+	WebDriver driver=null;
+
+	public void initialization() {
+		System.setProperty("webdriver.chrome.driver",
+				"C://chetan.phadke//Backup//Data from D//software//chromedriver_win32//chromedriver.exe");
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]:
```suggestion
				"C://chetan.phadke//Backup//Data from D//software//chromedriver_win32//chromedriver.exe");
				
				Here please add System path not hard coded path.
```

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+
+//Selenium Wrapper Class
+public class PredefinedMethods {
+	WebDriver driver=null;
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]:
```suggestion
	WebDriver driver=null;
	
	Its value is already null then why you assigned 'null' value.
```

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	public void initialization() {
+		System.setProperty("webdriver.chrome.driver",
+				"C://chetan.phadke//Backup//Data from D//software//chromedriver_win32//chromedriver.exe");
+		WebDriver driver = new ChromeDriver();
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]:
```suggestion
		WebDriver driver = new ChromeDriver();
		
		Earlier you already given WebDriver driver = null and again here you are giving return type of driver Why ?
```

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		driver.get("https://opensource-demo.orangehrmlive.com/");
+		driver.manage().window().maximize();
+		driver.manage().deleteAllCookies();
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]:
```suggestion
		driver.manage().deleteAllCookies();
```
Remove this line.

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		driver.manage().timeouts().pageLoadTimeout(30, TimeUnit.SECONDS);
+	}
+	
+	
+	public PredefinedMethods() {
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]:
```suggestion
	public PredefinedMethods() {
	
	What is the need of this method ?
```

File: src/main/java/com/technocredits/orangeHRM/util/PropertyFileReader.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+	public Properties initializePropertyFile(String propFileName) {
+		
+		File file=new File(propFileName+".properties");
+		try {
+			System.out.println("Null pointer");
+			FileInputStream input= new FileInputStream(file);
+			
+			prop=new Properties();
+			
+			prop.load(input);
```
Comment (krishnakananitest109) [inline src/main/java/com/technocredits/orangeHRM/util/PropertyFileReader.java]: Remove extra space in above.
Add comment what this method is actually doing.

File: src/test/java/com/technocredits/orangeHRM/testcases/CommonTest.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.technocredits.orangeHRM.testcases;
+
+import com.technocredits.orangeHRM.pages.LoginPage;
+
+public class CommonTest {
+
+		void login(String uname, String password) {
+			LoginPage loginpageInstance= LoginPage.getInstance();
+			//Xpath for username
```
Comment (krishnakananitest109) [inline src/test/java/com/technocredits/orangeHRM/testcases/CommonTest.java]:
```suggestion
			//Xpath for username
			
			Remove this type of comment.
```

Comment (chetanphadke3) [reply src/test/java/com/technocredits/orangeHRM/testcases/CommonTest.java]: Comments Removed

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java (hunk 5 above)
Comment (chetanphadke3) [reply src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]: No need of the constructor, removed it.

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java (hunk 4 above)
Comment (chetanphadke3) [reply src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]: Removed the delete cookies statement.

File: src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java (hunk 3 above)
Comment (chetanphadke3) [reply src/main/java/com/technocredits/orangeHRM/base/PredefinedMethods.java]: Okay. My bad. Deleted the return type webdriver.

File: src/main/java/com/technocredits/orangeHRM/util/PropertyFileReader.java (hunk 6 above)
Comment (chetanphadke3) [reply src/main/java/com/technocredits/orangeHRM/util/PropertyFileReader.java]: Extra spaces removed. And comments are added accordingly.

**Comment by chetanphadke3** [comment]
Adding files- PredefinedMethods, LoginConfigurationPath, ChromeDriverPath, LoginPage, PropertyFileReader, CommonTest and LoginTest with some comments. And addding chromedriver.exe in src/main/resources package.

**Pull request merged**